### PR TITLE
[CLI-190] Add "auto" registration support

### PR DESCRIPTION
**Pull request by codemodify**

### CODE REVIEW

File: business_logic/service.go [hunk 1]
```diff
@@ -23,6 +23,7 @@ type Service struct {
 	Secret     string        `json:"secret" mapstructure:"secret"`
 	Type       ServiceTypeID `json:"type" mapstructure:"type"`
 	UID        string        `json:"uid" mapstructure:"uid"`
+	ServiceID  string        `json:"serviceID" mapstructure:"serviceID"`
```
Comment (danawoodman) [inline business_logic/service.go]: Why is this here? This is redundant since `UID` *is* the `ServiceID`

File: autoregistration/autoregistration.go [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+				Disabled:   serviceConfig.Disabled,
+				HardwareID: deviceUniqueID,
+				Hostname:   serviceConfig.Hostname,
+				Name:       reg.UID,
+				Overload:   overload,
+				Port:       serviceConfig.Port,
+				Secret:     reg.Secret,
+				Type:       serviceConfig.Type,
+				UID:        reg.UID,
+				ServiceID:  serviceTemplateID,
```
Comment (danawoodman) [inline autoregistration/autoregistration.go]: Redundant

File: go.mod [hunk 3]
```diff
@@ -7,8 +7,9 @@ require (
 	github.com/briandowns/spinner v1.7.0
 	github.com/caio/go-tdigest v3.1.0+incompatible // indirect
 	github.com/codemodify/SystemKit v1.4.5
-	github.com/codemodify/systemkit-platform-cpu v1.2.0
-	github.com/denisbrodbeck/machineid v1.0.1
+	github.com/codemodify/systemkit-platform-cpu v1.3.2
+	github.com/codemodify/systemkit-platform-machineid v1.0.1
+	github.com/codemodify/systemkit-platform-os v1.0.0
```
Comment (danawoodman) [inline go.mod]: I need to understand why this was changes as it is a key component. I can't merge confidently until I understand this.

File: autoregistration/autoregistration.go [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+	"github.com/remoteit/cli/system"
+	"github.com/remoteit/cli/tools"
+)
+
+const registrationTimeout = 10 * time.Second
+
+// AutoRegister -
+func AutoRegister(registrationKey string) error {
+	if stringHelpers.IsNullOrEmpty(registrationKey) {
+		return fmt.Errorf("registrationKey (aka BIC) is invalid")
```
Comment (danawoodman) [inline autoregistration/autoregistration.go]: What is BIC?

File: autoregistration/autoregistration.go [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+	}
+
+	// see if we have a saved "deviceUniqueID"
+	deviceUniqueID := config.DeviceUniqueID()
```
Comment (danawoodman) [inline autoregistration/autoregistration.go]: Should  be called `UniqueDeviceID`  which reads more like  english I think

File: autoregistration/autoregistration.go [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+		if err != nil {
+			logger.Error("error service config: ", err)
+			return false, err
+		}
+		logger.Debugf("device config: %s", jsonHelpers.AsJSONString(serviceConfig))
+
+		// Check if we have this one already
+		existingServiceIndex := -1
+		for esInxed, es := range existingServices {
+			if es.ServiceID == serviceTemplateID {
```
Comment (danawoodman) [inline autoregistration/autoregistration.go]: Use `UID` instead so we don't need to redundant `ServiceID`

File: environment/machine.go [hunk 7]
```diff
@@ -2,28 +2,54 @@ package environment
...
 
-	if machineID == "" {
-		machineID = MacAddress()
-	}
+// GenerateNewID -
+func GenerateNewID(existingID string) string {
+	return fmt.Sprintf("%s_%s", existingID, stringHelpers.NewGUID())
+}
+
+// // ComputedUniqueID_Example -
```
Comment (danawoodman) [inline environment/machine.go]: Can we  remove these comments...?

File: autoregistration/autoregistration.go (hunk 5 above)
Comment (danawoodman) [reply autoregistration/autoregistration.go]: How about `SystemID` instead?

File: autoregistration/autoregistration.go (hunk 2 above)
Comment (codemodify) [reply autoregistration/autoregistration.go]: This is the serviceTemplateID for the service template (project) item being downloaded

File: autoregistration/autoregistration.go (hunk 4 above)
Comment (codemodify) [reply autoregistration/autoregistration.go]: fixed

File: autoregistration/autoregistration.go (hunk 5 above)
Comment (codemodify) [reply autoregistration/autoregistration.go]: done

File: environment/machine.go (hunk 7 above)
Comment (codemodify) [reply environment/machine.go]: renamed to "secure"
